plugin/metrics/prometheus/factory_test.go

```diff
@@ -55,17 +55,31 @@ func TestWithDefaultConfiguration(t *testing.T) {
 }
 
 func TestWithConfiguration(t *testing.T) {
-	f := NewFactory()
-	v, command := config.Viperize(f.AddFlags)
-	err := command.ParseFlags([]string{
-		"--prometheus.server-url=http://localhost:1234",
-		"--prometheus.connect-timeout=5s",
+	t.Run("With custom configuration and no space in token file path", func(t *testing.T) {
+		f := NewFactory()
+		v, command := config.Viperize(f.AddFlags)
+		err := command.ParseFlags([]string{
+			"--prometheus.server-url=http://localhost:1234",
+			"--prometheus.connect-timeout=5s",
+			"--prometheus.token-file=test/test_file.txt",
+		})
+		require.NoError(t, err)
+		f.InitFromViper(v, zap.NewNop())
+		assert.Equal(t, f.options.Primary.ServerURL, "http://localhost:1234")
+		assert.Equal(t, f.options.Primary.ConnectTimeout, 5*time.Second)
+		assert.Equal(t, f.options.Primary.TokenFilePath, "test/test_file.txt")
+	})
+	t.Run("With space in token file path", func(t *testing.T) {
+		f := NewFactory()
+		v, command := config.Viperize(f.AddFlags)
+		err := command.ParseFlags([]string{
+			"--prometheus.token-file=test/ test_file.txt",
+		})
+		require.NoError(t, err)
+		f.InitFromViper(v, zap.NewNop())
+		assert.Equal(t, f.options.Primary.TokenFilePath, "test/test_file.txt")
```

Comment:
ahh.. I am sorry i misunderstood your suggestion. I am modifying this test case to correctly depict required scenario.

Reply:
> This is okay, but what about the case where the user inputs `"test/test file.txt"`? I think the `f.options.Primary.TokenFilePath` should still stay as `"test/test file.txt"`, because it's perfectly valid (but not advised) to have spaces in a filename (you can try it out in `TestLoadToken`).

I removed stripWhiteSpace() from cfg.TokenFilePath = v.GetString(cfg.namespace + suffixTokenFilePath) which fixed this edge case